src/core/renderer.js

```diff
@@ -7,6 +9,19 @@ export class Renderer {
     // Abstract method
   }
 
+  static shouldRefreshChildFrameWithMorphing(parentFrame, currentFrame, newFrame) {
+    return currentFrame instanceof FrameElement &&
+      // newFrame cannot yet be an instance of FrameElement because custom
+      // elements don't get initialized until they're attached to the DOM, so
+      // test its Element#nodeName instead
+      newFrame instanceof Element && newFrame.nodeName === "TURBO-FRAME" &&
+      currentFrame.shouldReloadWithMorph &&
+      currentFrame.id === newFrame.id &&
```

Comment:
Isn't this ID check redundant? idiomorph won't match nodes with different IDs, right?

Reply:
Confirmed. We can ditch this as of Idiomorph v0.4

Reply:
We should now also be able to remove line 17, now that I think about it.

Reply:
@jorgemanrubia Ah, turns out we DO need these lines after all, because Turbo is reusing the same callback for both morphed AND removed hooks, and thus `newFrame` can be `undefined`:
```js
class DefaultIdiomorphCallbacks {
...
  beforeNodeRemoved = (node) => {
    return this.beforeNodeMorphed(node)
  }
```